src/Installer/DNS/Mac/DownloadDnsDock.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Dock\Installer\DNS\Mac;
+
+use Dock\Installer\InstallerTask;
+use Dock\IO\ProcessRunner;
+use Dock\IO\UserInteraction;
+use SRIO\ChainOfResponsibility\DependentChainProcessInterface;
+
+class DownloadDnsDock extends InstallerTask implements DependentChainProcessInterface
+{
+    /**
+     * @var UserInteraction
+     */
+    private $userInteraction;
+
+    /**
+     * @var ProcessRunner
+     */
+    private $processRunner;
+
+    /**
+     * @param UserInteraction $userInteraction
+     * @param ProcessRunner $processRunner
+     */
+    public function __construct(UserInteraction $userInteraction, ProcessRunner $processRunner)
+    {
+        $this->userInteraction = $userInteraction;
+        $this->processRunner = $processRunner;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function run()
+    {
+        $this->userInteraction->writeTitle("Pulling image tonistiigi/dnsdock");
+        $this->userInteraction->write("This could take a while when run for the first time");
+        $this->processRunner->run('docker pull tonistiigi/dnsdock');
```

Comment:
That won't work I think because the environment variables won't be setup when installing the system. Moreover, you are not sure that the machine is actually up. 

FYI you can have the environment variable definitions by running `docker-machine env [name]`, if it helps.